src/pykeen/regularizers.py

```diff
@@ -77,18 +85,29 @@ def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
         """Compute the regularization term for one tensor."""
         raise NotImplementedError
 
-    def update(self, *tensors: torch.FloatTensor) -> None:
+    def update(self, *tensors: torch.FloatTensor) -> bool:
         """Update the regularization term based on passed tensors."""
-        if self.apply_only_once and self.updated:
-            return
+        if not self.training or not torch.is_grad_enabled() or (self.apply_only_once and self.updated):
+            return False
         self.regularization_term = self.regularization_term + sum(self.forward(x=x) for x in tensors)
         self.updated = True
+        return True
 
     @property
     def term(self) -> torch.FloatTensor:
         """Return the weighted regularization term."""
         return self.regularization_term * self.weight
 
+    def pop_regularization_term(self) -> torch.FloatTensor:
+        """Return the weighted regularization term, and reset the regularize afterwards."""
+        # If there are tracked parameters, update based on them
+        if len(self.tracked_parameters) > 0:
```

Comment:
No `if self.tracked_parameters:`  abbreviation? 😉

Reply:
heh we could do that

Reply:
🤷 😄

Reply:
Fixed in 9eefd37f